### PR TITLE
[Merged by Bors] - p2p: add spread to the discovery advertisement interval

### PR DESCRIPTION
**Pull request by ivan4th**
## Motivation

Need to lessen DHT overload from the routing discovery mechanism

## Description

Default to 2h +/- 1h discovery advertisement interval

## Test Plan

Verified on a mainnet node

### CODE REVIEW

File: p2p/dhtdiscovery/discovery.go [hunk 1]
```diff
@@ -197,14 +205,15 @@ type Discovery struct {
...
-	advertiseInterval   time.Duration
-	advertiseRetryDelay time.Duration
-	findPeersRetryDelay time.Duration
+	timeout                 time.Duration
+	bootstrapDuration       time.Duration
+	minPeers, highPeers     int
+	backup, bootnodes       []peer.AddrInfo
+	advertiseDelay          time.Duration
+	advertiseInterval       time.Duration
+	advertiseIntervalSpread time.Duration
```
Comment (fasmat) [inline p2p/dhtdiscovery/discovery.go:214]: Instead of making this configurable would it make sense to instead have it a relative amount, e.g. 1%?

**Review by fasmat — APPROVED** [review]

**Comment by pigmej** [comment]
bors merge

**Comment by ivan4th** [comment]
bors merge

**Comment by ivan4th** [comment]
systests timed out with rather unclear reason, retrying

**Comment by ivan4th** [comment]
Timed out on this job: `CI / build-tools (go-spacemesh, m2-pro) (push)`

**Comment by ivan4th** [comment]
bors merge

**Comment by ivan4th** [comment]
Test flake on Windows:
```
    grpcserver_test.go:759: 
        	Error Trace:	D:/a/go-spacemesh/go-spacemesh/api/grpcserver/grpcserver_test.go:759
        	Error:      	An error is expected but got nil.
        	Test:       	TestSmesherService/PostSetupStatusStream
    logger.go:146: 2024-04-09T15:31:53.496Z	INFO	grpc	finished streaming call with code OK	{"grpc.start_time": "2024-04-09T15:31:53Z", "system": "grpc", "span.kind": "server", "grpc.service": "spacemesh.v1.SmesherService", "grpc.method": "PostSetupStatusStream", "peer.address": "127.0.0.1:65112", "grpc.code": "OK", "grpc.time_ms": 50.828}
    logger.go:146: 2024-04-09T15:31:53.498Z	INFO	grpc	stopping the grpc server
--- FAIL: TestSmesherService/PostSetupStatusStream (1.11s)
```

**Comment by fasmat** [comment]
bors merge

**Comment by ivan4th** [comment]
bors merge

**Pull request closed without merging**